### PR TITLE
fix(slack): Point to newer docs, better defaults

### PR DESCRIPTION
**Pull request by phyzical**
* added new default going forward uncommented
* added link to future guide
* added support for new signing-secret
* slight rejig to config

### CODE REVIEW

File: sentry/config.example.yml [hunk 1]
```diff
@@ -95,9 +95,12 @@ transaction-events.force-disable-internal-project: true
 #####################
 
 # Refer to https://forum.sentry.io/t/how-to-configure-slack-in-your-on-prem-sentry/3463 for setup instructions.
+# Or https://develop.sentry.dev/integrations/slack/
```
Comment (BYK) [inline sentry/config.example.yml:98]: Let's make this the default link :)

File: sentry/config.example.yml [hunk 2]
```diff
@@ -95,9 +95,12 @@ transaction-events.force-disable-internal-project: true
...
 # slack.client-id: <'client id'>
 # slack.client-secret: <client secret>
+# slack.signing-secret: <signing secret>
+## if you made your slack bot before july 2020 set legacy-app to True
+## and use verfication-token instead of signing-secret
 # slack.verification-token: <verification token>
-## only uncomment legacy-app if you made your slack bot after july 2020
-# slack.legacy-app: False
\ No newline at end of file
+slack.legacy-app: False
```
Comment (BYK) [inline sentry/config.example.yml:106]: I'd move this above the verification token or reword the comment so the ordering matches (first token, then legacy-app)

File: sentry/config.example.yml (hunk 1 above)
Comment (phyzical) [reply sentry/config.example.yml]: done :)

File: sentry/config.example.yml (hunk 2 above)
Comment (phyzical) [reply sentry/config.example.yml]: how about that?

Comment (BYK) [reply sentry/config.example.yml]: Nice :)

**Pull request merged**